frameworks/hdfs/tests/test_ssl_kerberos_auth.py

```diff
@@ -0,0 +1,261 @@
+        "service": {
+            "name": config.SERVICE_NAME,
+            "service_account": service_account,
+            "service_account_secret": service_account,
+            "security": {
+                "kerberos": {
+                    "enabled": True,
+                    "kdc": {
+                        "hostname": kerberos.get_host(),
+                        "port": int(kerberos.get_port())
+                    },
+                    "realm": kerberos.get_realm(),
+                    "keytab_secret": kerberos.get_keytab_path(),
+                },
+                "transport_encryption": {
+                    "enabled": True
+                }
+            }
+        },
+        "hdfs": {
+            "security_auth_to_local": auth.get_principal_to_user_mapping()
+        }
+    }
+
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+    try:
+        sdk_install.install(
+            config.PACKAGE_NAME,
+            config.SERVICE_NAME,
+            config.DEFAULT_TASK_COUNT,
+            additional_options=service_kerberos_options,
+            timeout_seconds=30 * 60)
+
+        yield {**service_kerberos_options, **{"package_name": config.PACKAGE_NAME}}
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.fixture(scope='module', autouse=True)
+def hdfs_client(kerberos, hdfs_server):
+    try:
+        client_id = "hdfs-client"
+        client = {
+            "id": client_id,
+            "mem": 1024,
+            "user": "nobody",
+            "container": {
+                "type": "MESOS",
+                "docker": {
+                    "image": "elezar/hdfs-client:dev",
+                    "forcePullImage": True
+                },
+                "volumes": [
+                    {
+                        "containerPath": "/hadoop-2.6.0-cdh5.9.1/hdfs.keytab",
+                        "secret": "hdfs_keytab"
+                    }
+                ]
+            },
+            "secrets": {
+                "hdfs_keytab": {
+                    "source": kerberos.get_keytab_path()
+                }
+            },
+            "networks": [
+                {
+                    "mode": "host"
+                }
+            ],
+            "env": {
+                "REALM": kerberos.get_realm(),
+                "KDC_ADDRESS": kerberos.get_kdc_address(),
+                "JAVA_HOME": "/usr/lib/jvm/default-java",
+                "KRB5_CONFIG": "/etc/krb5.conf",
+                "HDFS_SERVICE_NAME": config.SERVICE_NAME,
+            }
+        }
+
+        sdk_marathon.install_app(client)
+
+        auth.write_krb5_config_file(client_id, "/etc/krb5.conf", kerberos)
+        dcos_ca_bundle = auth.fetch_dcos_ca_bundle(client_id)
+
+        yield {**client, **{"dcos_ca_bundle": dcos_ca_bundle}}
+
+    finally:
+        sdk_marathon.destroy_app(client_id)
+
+
+# TODO(elezar) Is there a better way to determine this?
+DEFAULT_JOURNAL_NODE_TLS_PORT = 8481
```

Comment:
Parsing the package description is the only way that comes to mind immediately.

Reply:
I think we could also download `hdfs-site.xml` and extract the ports that way.